apps/aetestframework/lib/worker.ex

```diff
@@ -1,599 +1,161 @@
-      File.rm_rf(path <> "aewallet_#{val.port}")
-      File.rm_rf(path <> "peerkeys_#{val.port}")
-      File.rm_rf(path <> "rox_db_#{val.port}")
+      File.rm_rf(path <> "aewallet_#{port}")
+      File.rm_rf(path <> "peerkeys_#{port}")
+      File.rm_rf(path <> "rox_db_#{port}")
     end)
 
     {:reply, :ok, %{}}
   end
 
-  def handle_call({:delete_node, node_name}, _, state) do
-    if Map.has_key?(state, node_name) do
-      Port.command(state[node_name].process_port, ":erlang.halt()\n")
-      Port.close(state[node_name].process_port)
-      new_state = Map.delete(state, node_name)
-      {:reply, :ok, new_state}
-    else
-      {:reply, :unknown_node, state}
-    end
-  end
-
-  def handle_call({:ports_info}, _, state) do
-    ports_info = for {name, _} <- state, do: Port.info(state[name].process_port)
-    {:reply, ports_info, state}
-  end
-
-  def handle_call(
-        {:get_oracle_interaction_objects, node},
-        _,
-        state
-      ) do
-    {:reply, state[node].oracle_interaction_objects}
+  def handle_info(_data, state) do
+    {:noreply, state}
   end
 
-  def handle_call({:get_top_block, node}, _, state) do
-    {:reply, state[node].top_block, state}
+  def process_result(result) do
+    filtered_result = String.replace(result, "\n", "")
+    matched_result = Regex.run(~r/\"(.*?)\"/, filtered_result)
+    base_decoded = Base.decode32!(List.last(matched_result))
+    :erlang.binary_to_term(base_decoded)
   end
 
-  def handle_call({:get_top_block_hash, node}, _, state) do
-    {:reply, state[node].top_block_hash, state}
+  defp receive_result(key, fun) do
+    receive do
+      {_port, {:data, result}} ->
+        if result =~ key do
+          fun.(result)
+        else
+          receive_result(key, fun)
+        end
+    end
   end
 
-  def handle_call({:new_node, node_name, iex_num}, _, state) do
-    cond do
-      Map.has_key?(state, node_name) ->
-        {:reply, :already_exists, state}
-
-      busy_port?("300#{iex_num}") || busy_port?("400#{iex_num}") ->
-        {:reply, :busy_port, state}
-
-      true ->
-        # Running the new elixir-node using Port
-        path = String.replace(System.cwd(), ~r/(?<=elixir-node).*$/, "")
-        process_port = Port.open({:spawn, "make iex-node NODE_NUMBER=#{iex_num}"}, [:binary, cd: path])
-        receive_result(state)
-        port = String.to_integer("400#{iex_num}")
-        sync_port = String.to_integer("300#{iex_num}")
-
-        new_state =
-          Map.put(state, node_name, %{
-            process_port: process_port,
-            port: port,
-            sync_port: sync_port,
-            top_block: nil,
-            top_block_hash: nil,
-            miner_pubkey: nil,
-            miner_balance: 0,
-            peers: %{},
-            oracle_interaction_objects: %{}
-          })
-
-        {:reply, :ok, new_state}
-    end
+  @doc """
+  Checking if the port is busy
+  """
+  def busy_port?(port) do
+    :os.cmd('lsof -i -P -n | grep -w #{port}') != []
```

Comment:
lsof should be added to dependencies, with reference to the test framework

Reply:
`lsof` is a unix command, what do you mean by adding it to the dependencies?

Reply:
lsof needs to be installed on the system this is supposed to run on

Reply:
I can add this to the README.md file, that it is required in order for the Multinode sync test Framework to work.

Reply:
Yes